python-lib/osisoft_client.py

```diff
@@ -896,6 +918,88 @@ def close(self):
     # }
 
 
+class OSIsoftBatchWriter(object):
+    # Each row of data added (write_row) first goes in a streak buffer
+    # streak buffer is meant to be used to push a flow of values to one AF path / webid
+    # If the new row concerns another webid, the current streak buffer is flushed into the request buffer
+    # When request buffer is full, it is flushed into the batch endpoint
+
+    # Each write_row call is adds an entry to the responses list
+    # The pointer into that list is past as a _dku_counter parameter, extracted from the requests just before sending the batch
+    # Uppon receiving the batch response, each individual reponse is reordered
+    #   and the status code / error messages are stored into the responses list at the right pointer
+
+    # Possible improvement: flush the responses list (and result writing in the recipe.py) at each _flush_requests to keep memory from going up.
+
+    # If you have to review this, please accept my sincere apologies.
+
+    def __init__(self, client, max_streak_buffer_size=500, max_requests_buffer_size=500):
+        logger.info("Initializing OSIsoftBatchWriter, msbs={}, mrbs={}".format(max_streak_buffer_size, max_requests_buffer_size))
+        self.client = client
+        self.streak_buffer = []     # list of points from a single webid -> can be all sent in one request
+        self.requests_buffer = []   # list of independant requests -> sent one batch at a time
+        self.responses = []         # building a list of reponses in same order as write_row calls
+        self.current_webid = None
+        self.max_buffer_size = max_streak_buffer_size
+        self.max_requests_buffer_size = max_requests_buffer_size
+        self.current_streak = 0
+        self.row_number = 0
+
+    def write_row(self, webid, timestamp, value):
+        response = None
+        timestamp = self._timestamp_convertion(timestamp)
```

Comment:
This is silently using a different timestamp if the timestamp for the current row is not specified. Should it be logged or specified in some way ? Isn't it risky to decide this without warning or control ?
(comment refering to the implem of timestamp_convertion method)